stats/stats.go

```diff
@@ -539,3 +539,46 @@ func RoundToDigits(v float64, digits int) float64 {
 func Round(v float64) float64 {
 	return RoundToDigits(v, 4)
 }
+
+// Occurrence is a type that stores the occurrence of the key.
+type Occurrence struct {
+	ipUsage map[string]int
+}
+
+// NewOccurrence create a new occurrence map.
+func NewOccurrence() *Occurrence {
+	o := new(Occurrence)
+	o.ipUsage = make(map[string]int)
+
+	return o
+}
+
+// Record records a key value pair.
+func (m *Occurrence) Record(key string) {
+	m.ipUsage[key]++
+}
+
+// PrintAndAggregate print the ip usage and aggregate to the total count map.
+func (m *Occurrence) PrintAndAggregate(ipCountMap map[string]int) string {
```

Comment:
shouldn't it take another Occurence rather than raw map? (not sure, just wondering) - also maybe uint and/or int64 and/or uint64

Reply:
given we already had IPCountMap I guess it's fine, even though somewhat awkward  api to combine a String() and a Transfer()/aggregate style into one, but it does save a loop